### PR TITLE
fixes a test and some text

### LINKED ISSUE #2256: Not Properly Checking a Step
Challenge http://freecodecamp.com/challenges/waypoint-generate-random-whole-numbers-with-javascript has an issue. Please describe how to reproduce it, and include links to screenshots if possible.

One of the steps requires us to multiply by 10. I didn't do this part (tried doing it without looking) and it still passes me. Possible bug I think.

### LINKED ISSUE #2819: bug Basic Javascript 30
Challenge http://www.freecodecamp.com/challenges/waypoint-generate-random-whole-numbers-with-javascript has an issue. Please describe how to reproduce it, and include links to screenshots if possible.

I typed in return(Math.floor(Math.random()*10));
without the *10, to figure out exactly what Math.floor does  (I'm still not sure) and I still cleared the waypoint, Eventhough that's not what you're supposed to do.

### PR DESCRIPTION
**Pull request by bugron**
FIxes #2819.
Fixes #2256.

### CODE REVIEW

**Comment by benmcmahon100** [comment]
Hey @bugron please avoid removing tests unless you're positive that they aren't needed. Some of the tests are there to prevent extreme edge cases but it's better to have more than less as long as the errors they have are clear

**Comment by ahstro** [comment]
Instead of creating a new commit to revert something you did in a previous one, you could use `git commit --amend` in the future. Would you mind [rebasing](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request) this to remove the extra commit? It keeps the git history cleaner.

**Comment by bugron** [comment]
@ahstro done. Thanks.

File: seed/challenges/basic-javascript.json [hunk 1]
```diff
@@ -895,21 +895,21 @@
...
         "Let's give this technique a go now."
       ],
       "tests":[
-        "assert(typeof(myFunction()) === \"number\", 'The result of myFunction should be a number');",
-        "assert(editor.getValue().match(/Math.random/g), 'You should be using Math.random to create a random number');",
-        "assert(!(''+myFunction()).match(/\\./g), 'You should have multiplied the result of Math.random by 10 to make it a number that\\'s greater than zero');",
-        "assert(editor.getValue().match(/Math.floor/g), 'You should use Math.floor to remove the decimal part of the number');"
+        "assert(typeof(myFunction()) === \"number\", 'The result of <code>myFunction</code> should be a number.');",
+        "assert(editor.getValue().match(/Math.random/g), 'You should be using Math.random to create a random number.');",
+        "assert(editor.getValue().match(/Math.random\\s*?\\(\\s*?\\)\\s*?\\*\\s*?10/g), 'You should have multiplied the result of <code>Math.random</code> by 10 to make it a number that\\'s between zero and nine.');",
```
Comment (ahstro) [inline seed/challenges/basic-javascript.json]: I'm not sure this regex is the best way to check if the answer is correct for two reasons:
 1. `return Math.floor(Math.random() * 100);` passes. This is the easiest to solve since you could just add the closing bracket to the regex.
 2. This is the one I'm really worried about. `return Math.floor(10 * Math.random());` doesn't pass. So you'd either have to write a really long and inconvenient regex that checks whether `10` is before or after `Math.random()` or find some other way to check the solution.

**Comment by bugron** [comment]
@ahstro I've updated the code.

**Comment by ahstro** [comment]
Hmm.. from the perspective of a JS beginner or early programmer, I guess this works; from the perspective of someone who's meticulous and enjoys finding holes in code, there are still a few problems (`10 * Math.random() * 10` passes, `5 * 2 * Math.random()` doesn't), but I'm sure I'm digging too deep now :wink: 
Could someone with more authority on the matter make the decision whether to merge this or not? (*cough* @FreeCodeCamp/owners *cough*)

Small note on the regex: I don't think the `?`s are necessary after the `*`s since the latter means "0 or more".

**Comment by bugron** [comment]
@ahstro are you serious? One can solve this challenge in a billion ways (and not only this challenge), so should I consider all of these solutions? Of course no. 
And about `\\s*?`. This is a recommendation form @benmcmahon100.

**Comment by ahstro** [comment]
That's my point, what really needs to be fixed is how the tests are executed, there should be a variable we can check the value of, not how the value is calculated. I believe I've heard some talk about changing how tests work, so this is probably valid until then. Sorry if I'm coming off as nitpicky :)

Could you explain why that's necessary @benmcmahon100?

**Comment by benmcmahon100** [comment]
The reason why the method is being evaluated is because that's what the waypoints are meant to teach. As for *5*2 vs *10 if i asked you to convert dollars to cents you'd multiply it by ten. Doing 5*2 or (10^2)/10/5*5-15 is just unnecessary. The other major issue is that i can spend a day making one infallible challenge or twenty quite good ones so the trade off of having to use the syntax we want is an easy one.

**Comment by benmcmahon100** [comment]
@bugron There's no test for Math.floor and the test for 10 * Math.random() is not really needed but it can e left in. People should be using the syntax they are given in the description but with leeway on the spaces

**Comment by ahstro** [comment]
@benmcmahon100 I think "people should be using the syntax they are given" is quite a dull attitude. I remember using Codecademy back in the day, and something that really helped me grasp programming and learn how a language worked was the many different ways I could write something to get the same result. For a beginner or early programmer it might be hard to understand why `x = 10 * 2` works, but `10 * 2 = x` doesn't, so making our tests as close to actual JavaScript as possible feels quite important to me. This is obviously hard for us to do, especially with regex, but I just wanted to explain myself.

**Comment by bugron** [comment]
@BerkeleyTrue thanks. Working on https://github.com/FreeCodeCamp/FreeCodeCamp/pull/3083, https://github.com/FreeCodeCamp/FreeCodeCamp/pull/3084 and https://github.com/FreeCodeCamp/FreeCodeCamp/pull/3086.
